tests/comparisons/Model/testBakeAssociationDetectionCategoriesProductsTable.php

```diff
@@ -47,6 +47,25 @@ public function initialize(array $config)
         ]);
     }
 
+    /**
+     * Default validation rules.
+     *
+     * @param \Cake\Validation\Validator $validator Validator instance.
+     * @return \Cake\Validation\Validator
+     */
+    public function validationDefault(Validator $validator)
+    {
+        $validator
+            ->integer('category_id')
+            ->allowEmpty('category_id', 'create');
```

Comment:
Is the difference with this implementation vs the old one that keys can be blank on create?

Reply:
looks for `null` able column validation is not necessary , but for `not null` it is important